src/Commands/new.js

```diff
@@ -2,88 +2,238 @@
+const fs = require('fs-extra')
+const BaseCommand = require('./Base')
+const exec = require('child_process').exec
+
+/**
+ * Command that create a new AdonisJs application.
+ *
+ * @class New
+ * @module Adonis
+ * @submodule cli
+ * @extends Base
+ */
+class New extends BaseCommand {
+
+  /**
+   * Returns the signature of the command.
+   *
+   * @readOnly
+   * @property signature
+   * @return {string}
+   */
+get signature () {
+  return `new {name:Name of your application}
+    {--skip-install?:Skip the installation process}
+    {--branch?=@value:Branch to be used}
+    {--blueprint?=@value:Repository to be used}
+    {--yarn?:Use yarn to install dependencies}
+    {--npm?:Use npm to install dependencies}
+  `
+}
+
+  /**
+   * Returns the description of the command.
+   *
+   * @readOnly
+   * @property description
+   * @return {string}
+   */
+  get description () {
+    return 'Scaffold a new AdonisJs application with the name provided.'
+  }
+
+  /**
+   * Scaffold a new AdonisJs application by following these steps.
+   *
+   *  1. Check the Node.js version to ensure that it match the requirements.
+   *  2. Verify that the application folder doesn't already exists.
+   *  3. Clone the blueprint & branch requested to the applicatin folder.
+   *  4. Install all dependencies using yarn or npm
+   *  5. Remove the .git folder to have a clean install.
+   *  6. Copy the default .env.example to .env
+   *  7. Generate a default APP_KEY using ace.
+   *
+   * @method handle
+   * @param  {array} args
+   * @param  {object} options
+   * @return {void}
+   */
+  * handle (args, options) {
+    this.branch = options.branch || 'master'
+    this.blueprint = options.blueprint || 'adonisjs/adonis-app'
+    this.applicationName = args.name
+    this.applicationPath = path.join(process.cwd(), args.name)
+    this.mustUse = null
+
+    if (options.npm) {
+      this.mustUse = 'npm'
+    } else if (options.yarn) {
+      this.mustUse = 'yarn'
+    }
+
+    this._dumpAsciiLogo()
+    this._checkRequirements()
+    yield this._verifyApplicationDoesntExist()
+    yield this._cloneRepository()
+
+    if (options['skip-install'] === null) {
+      yield this._installDependencies()
+    }
+
+    yield this._cleanProjectDirectory()
+    yield this._copyEnvironmentFile()
+
+    if (options['skip-install'] === null) {
+      yield this._generateSecureKey()
+    }
+
+    this.log()
+    this.success(`${this.icon('success')} Your application is ready!`)
+    this.log()
+
+    this.info(`${this.icon('info')} Follow below instructions to get started`)
+    this.log(`$ cd ${this.colors.magenta.bold(args.name)}`)
+    this.log(`$ npm run serve:dev`)
+    this.log()
+  }
 
-const repo = 'https://github.com/adonisjs/adonis-app.git'
-let branch = 'master'
+  /**
+   * Verify that the application folder doesn't already exists.
+   *
+   * @private
+   * @method _verifyApplicationDoesntExist
+   * @return {void}
+   */
+  * _verifyApplicationDoesntExist () {
```

Comment:
Make sure to listen for errors and ignore if error.code is `ENOENT`. For example the directory does not have write permission, which will throw an exception.

## How to reproduce?

1. Create a directory called `foo`.
2. Changes its permissions to `chmod 444 foo`
3. Now create the project `adonis new foo/bar`